src/getLoops.js

```diff
@@ -0,0 +1,58 @@
+const cytoscape = require('cytoscape');
+const cytoscapeAllPaths = require('cytoscape-all-paths');
+const _ = require('lodash');
+cytoscape.use(cytoscapeAllPaths);
+
+function getLoops(blueprint) {
+  const { nodes } = blueprint.blueprint_spec;
+  const cy = cytoscape();
+  let count = 0;
+  nodes.forEach((node) => cy.add({ data: { id: node.id, type: node.type } }));
```

Comment:
não daria para fazer esses 2 loops em uma passada só?

Reply:
Não dá pois pra criar um grafo no cytoscape ou eu adiciono todos os nodes e edges em um `cy.add({})` único ou adiciono primeiro os nodes e depois os edges.. Então optei pela segunda forma, já que pra fazer tudo em um `cy.add({})` eu precisaria fazer uma iteração prévia guardando nodes e edges em um array ou algo do tipo e ficaria um pouco mais complexo

Reply:
Ok, acho que por agora e considerando o tamanho das BPs é pouco relevante, o impacto de performance deve ser mínimo.
Talvez pensar depois em refatorar para durante a iteração vc guarda em uma variável e depois injeta no cy (se for possível).